net/host.go

```diff
@@ -81,8 +80,8 @@ type Config struct {
 
 // NewHost returns a new host
 func NewHost(cfg *Config) (*host, error) {
-	if cfg.KeyFile == "" {
-		cfg.KeyFile = defaultKeyFile
+	if cfg.DataDir == "" || cfg.KeyFile == "" {
+		panic("required parameters not set")
```

Comment:
I made sure you can't force a panic by doing something silly like `--data-dir ""` in the options processing.